grails-app/controllers/org/pih/warehouse/api/MobileProductApiController.groovy

```diff
@@ -150,4 +156,70 @@ class MobileProductApiController extends BaseDomainApiController {
         List products = productService.searchProducts(terms, [])
         render([data: products?.unique()] as JSON)
     }
+
+    /**
+    * Update a product's barcode (UPC) value.
+    * Endpoint: PUT /mobile/products/{id}/barcode
+    * Request body: { "upc": "123" }
+    */
+    @Transactional
+    def updateBarcode() {
+        def product = Product.get(params.id)
+        if (!product) {
+            render(status: 404, text: "Product was not found")
+            return
+        }
+
+        try {
+            def requestBody = request.JSON
+            def newUpc = requestBody?.upc?.trim()
+            def oldUpc = product.upc
+
+            if (!newUpc) {
+                render(status: 400, text: "Missing 'upc' in request body")
+                return
+            }
+
+            User user = session?.user
+            if (!user) {
+                render(status: 401, text: "Active user session required")
+                return
+            }
+
+            if (!(userService.hasRoleProductManager(user) || userService.isUserAdmin(user))) {
+                render(status: 403, text: "Insufficient privileges")
+                return
+            }
+
+            if (oldUpc?.trim() == newUpc) {
```

Comment:
Is this check needed? Do we care if there was an actual change or not?

Reply:
@olewandowski1 The reason I asked was because I ~was not sure about~ i don't like `status: 'no_change'` approach (since this is not our convention in such cases, and is not idiomatic REST), and because I was not fully sure how we'd like to have it done instead (if it should be a 204 in this case or a 200 with additional info), I thought it is just not required now, and we can get rid of it for now and revisit later if needed. On the other hand, if you just don't want to send the notification, you could check this on the notification send level.